src/wpool/mongoose_wpool_rdbms.erl

```diff
@@ -0,0 +1,54 @@
+-module(mongoose_wpool_rdbms).
+-behaviour(mongoose_wpool).
+
+-export([init/0]).
+-export([start/4]).
+-export([stop/2]).
+
+init() ->
+    case ets:info(prepared_statements) of
+        undefined ->
+            Heir = case whereis(ejabberd_sup) of
+                       undefined -> [];
+                       Pid -> [{heir, Pid, undefined}]
+                   end,
+            ets:new(prepared_statements,
+                    [named_table, public, {read_concurrency, true} | Heir]),
+            ok;
+        _ ->
+            ok
+    end.
+
+start(Host, Tag, WpoolOpts, RdbmsOpts) ->
+    try do_start(Host, Tag, WpoolOpts, RdbmsOpts)
+    catch
+        Err -> {error, Err}
+    end.
+
+do_start(Host, Tag, WpoolOpts0, RdbmsOpts) ->
+    Backend =
+        case lists:keyfind(server, 1, RdbmsOpts) of
+            {_, ConnStr} when is_list(ConnStr) -> odbc;
+            {_, Tuple} when is_tuple(Tuple) -> element(1, Tuple)
+        end,
+
+    try mongoose_rdbms_backend:backend_name() of
+        Backend -> ok;
+        OtherBackend ->
+            throw(#{reason => "Cannot start an RDBMS connection pool: only one RDBMS backend can be used",
```

Comment:
Hmm, I'm not sure if it's needed at all but I think it'd be good to allow connecting to different databases. Was it possible so far?

Reply:
No, it wasn't possible because of the use of `mongoose_rdbms_backend` proxy module, which is exactly one of the backends. (Also, `rdbms_server_type`)